src/CrectoAdmin/auto_resource.cr

```diff
@@ -0,0 +1,188 @@
+      pri_columns = [] of CrectoAdmin::Column
+      @columns.each do |column|
+        column_key = column.column_key.downcase
+        pri_columns << column if column_key.includes?("pri")
+      end
+      return nil if pri_columns.empty?
+      pri_columns.each do |column|
+        return column if column.column_name.includes? "id"
+      end
+      pri_columns[0]
+    end
+
+    def has_primary_key_auto?
+      pri_column = primary_key_column
+      return false if pri_column.nil?
+      pri_column = pri_column.as(CrectoAdmin::Column)
+      pri_column.extra.includes?("auto_increment")
+    end
+
+    def has_more_columns?
+      return false if primary_key_column.nil?
+      @columns.each do |column|
+        next if column == primary_key_column
+        next if column.column_type.nil?
+        return true
+      end
+      false
+    end
+  end
+
+  def self.build_header(uri)
+    String.build do |s|
+      s << "require \"mysql\"\n"
+      s << "require \"crecto-admin\"\n\n"
+      s << "module Repo\n"
+      s << "  extend Crecto::Repo\n"
+      s << "  config do |conf|\n"
+      s << "    conf.adapter = Crecto::Adapters::Mysql\n"
+      s << "    conf.uri = \"" << uri << "\"\n"
+      s << "  end\n"
+      s << "end\n\n"
+      s << "init_admin()\n\n"
+    end
+  end
+
+  def self.build_class(table)
+    return "# skip table: #{table.table_name}, no primary_key\n\n" if table.primary_key_column.nil?
+    return "# skip table: #{table.table_name}, no adaptable columns\n\n" unless table.has_more_columns?
+    String.build do |s|
+      s << "class #{table.class_name} < Crecto::Model\n"
+      s << "  set_created_at_field nil\n" unless table.has_created_at?
+      s << "  set_updated_at_field nil\n" unless table.has_updated_at?
+      s << "  schema \"#{table.table_name}\" do\n"
+      columns_symbols = [] of String
+      table.columns.each do |column|
+        if column == table.primary_key_column
+          s << "    field :#{column.column_name}, PkeyValue, primary_key: true\n"
+          columns_symbols << ":#{column.column_name}"
+        elsif column.is_created_at? || column.is_updated_at?
+          next
+        elsif column.column_type.nil?
+          s << "    # skip column: #{column.column_name} (#{column.data_type})\n"
+        else
+          s << "    field :#{column.column_name}, #{column.column_type}\n"
+          columns_symbols << ":#{column.column_name}"
+        end
+      end
+      s << "  end\n"
+      unless table.has_primary_key_auto?
+        s << "  def self.form_attributes\n"
+        s << "    [#{columns_symbols.join(", ")}]\n"
+        s << "  end\n"
+      end
+      s << "end\n"
+      s << "admin_resource(#{table.class_name}, Repo)\n\n"
+    end
+  end
+
+  def self.build_tail
+    String.build do |s|
+      s << "Kemal::Session.config do |config|\n"
+      s << "  config.secret = \"my super secret\"\n"
+      s << "end\n"
+      s << "Kemal.run\n\n"
+    end
+  end
+end
+
+def self.auto_script(repo)
```

Comment:
Do you know if this will also work for postgres / sqlite ?

Reply:
No, I do not think so. I think it only works for mysql. I did not put this in readme. Inside of the example, I stated that it only works for mysql. Even for mysql, the generated admin may not work for all tables, and needs to be reviewed. I view it just a helper for users that do not want to type code at start. And the our admins here seems really want this kind of helper.